exams/pearson/download.py

```diff
@@ -0,0 +1,251 @@
+"""Pearson SFTP download implementation"""
+import logging
+import os
+import zipfile
+from contextlib import contextmanager
+
+from django.conf import settings
+from paramiko import SSHException
+
+from exams.pearson.constants import (
+    EAC_SUCCESS_STATUS,
+    PEARSON_FILE_TYPE_EAC,
+    PEARSON_FILE_TYPE_VCDC,
+    VCDC_SUCCESS_STATUS,
+)
+from exams.pearson.exceptions import RetryableSFTPException
+from exams.pearson.readers import (
+    EACReader,
+    VCDCReader,
+)
+from exams.pearson import utils
+from exams.models import (
+    ExamAuthorization,
+    ExamProfile,
+)
+
+log = logging.getLogger(__name__)
+
+
+@contextmanager
+def locally_extracted(zip_file, member):
+    """
+    Context manager for temporarily extracting a zip file member to the local filesystem
+
+    Args:
+        zip_file (zipfile.ZipFile): the zip file to extract from
+        member (str): the name of the file to extract
+
+    Yields:
+        file: the extracted file object
+    """
+    extracted_file_path = zip_file.extract(member, path=settings.EXAMS_SFTP_TEMP_DIR)
+    try:
+        # csv.reader requires files to be opened in text mode, not binary
+        with open(extracted_file_path, 'r') as extracted_file:
+            yield extracted_file
+    finally:
+        if os.path.exists(extracted_file_path):
+            os.remove(extracted_file_path)
+
+
+class ArchivedResponseProcessor:
+    """
+    Handles fetching and processing of files stored in a ZIP archive on Pearson SFTP
+    """
+    def __init__(self, sftp):
+        self.sftp = sftp
+
+    def fetch_file(self, remote_path):
+        """        Fetches a remote file and returns the local path
+
+        Args:
+            remote_path (str): the remote path of the file to fetch
+
+        Returns:
+            str: the local path of the file
+        """
+        local_path = os.path.join(settings.EXAMS_SFTP_TEMP_DIR, remote_path)
+
+        self.sftp.get(remote_path, localpath=local_path)
+
+        return local_path
+
+    def filtered_files(self):
+        """
+        Walks a directory and yields files that match the pattern
+
+        Args:
+            pattern (re): the regex pattern to match the filename on
```

Comment:
Can you remove this arg from the docstring since there's no corresponding argument?